google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java

```diff
@@ -271,7 +272,38 @@ private boolean runningOnAppEngine() {
       return true;
```

Comment:
The code:
```
     if (getEnvEquals("GAE_RUNTIME", "java8")) {
       return true;
     }
     if (getEnvEquals("GAE_VM", "true")) {
       return true;
    }
```
should be:
```
     if (getEnvEquals("GAE_RUNTIME", "java8")) {
       return false;
     }
     if (getEnvEquals("GAE_VM", "true")) {
       return false;
    }
```

Reply:
Is the default case to return false? Should I just strike these lines?

Reply:
No, because we don't want to fall into the lines below.